File: backend/src/main/java/diaconat_mulhouse/fr/backend/core/converters/Converter.java
package diaconat_mulhouse.fr.backend.core.converters;

/**
 * Interface that convert DTO to Model and reverse
 *
 * @param <DTO>
 * @param <Model>
 * @author Arthur MATHIS - arthur.mathis@diaconat-mulhouse.fr
 */
public interface Converter<DTO, Model> {

    /**
     * Public method that converts a DTO to a Model
     *
     * @param dto The DTO
     * @return The new Model
     */
    public Model fromDto(DTO dto);

    /**
     * Public method that converts a Model to a DTO
     *
     * @param model The Model
     * @return The new DTO
     */
    public DTO toDto(Model model);

}

File: frontend/src/app/domain/converters/converter.ts
/**
 * @interface Converter
 * @author Arthur MATHIS <arthur.mathis@diaconat-mulhouse.fr>
 */
interface Converter<DTO, Model> {

    /**
     * Public method that converters a DTO into a Model
     * 
     * @param {DTO} dto The DTO
     * @returns {Model} The Model
     */
    fromDto(dto: DTO): Model;

    /**
     * Public method that converts a Model into a DTO
     * 
     * @param {Model} model The Model
     * @returns {DTO} The DTO
     */
    toDto(model: Model): DTO;

}

export type { Converter };